### PR TITLE
Feature/pointers qs

### PR DESCRIPTION
**Pull request by dkalathil**

### CODE REVIEW

File: src/algorithms/controllers/quickSort.js [hunk 1]
```diff
@@ -31,21 +31,44 @@ export default {
...
+
       chunker.add(5);
+
       chunker.add(11);
       chunker.add(12);
       const pivot = a[right];
-      chunker.add(6);
+      chunker.add(6, (vis, p, i1, j1) => {
+        vis.array.assignVariable('p', p);
+        if (i >= 0) {
```
Comment (Tianlalu233) [inline src/algorithms/controllers/quickSort.js]: bug exists
```suggestion
        if (i1 >= 0) {
```

File: src/algorithms/controllers/quickSort.js [hunk 2]
```diff
@@ -31,21 +31,44 @@ export default {
...
         chunker.add(8);
         do {
           j -= 1;
         } while (i <= j && pivot < a[j]);
-        chunker.add(9);
+        chunker.add(9, (vis, i1, j1) => {
+          if (i1 >= 0) {
+            vis.array.assignVariable('i', i1);
+          }
+          vis.array.assignVariable('j', j1);
```
Comment (Tianlalu233) [inline src/algorithms/controllers/quickSort.js]:
```suggestion
          if (j1 >= 0) {
            vis.array.assignVariable('j', j1);
          }
```

File: src/algorithms/controllers/quickSort.js (hunk 1 above)
Comment (dkalathil) [reply src/algorithms/controllers/quickSort.js]: fixed

File: src/algorithms/controllers/quickSort.js (hunk 2 above)
Comment (dkalathil) [reply src/algorithms/controllers/quickSort.js]: fixed

**Review by Tianlalu233 — APPROVED** [review]

**Pull request merged**